### PR TITLE
Setup unit tests

### PR DESCRIPTION
**Pull request by maschwenk**
Sets up unit tests for the configuration item. Hopefully open door to more basic unit testing down the line. Also slightly changes implementation of some previous code (#23) after noticing some bugs when running the unit tests 😄.

Also sets up:
1. A rake file
1. A contributing guide
1. Runs the unit tests in CI

### CODE REVIEW

**Comment by maschwenk** [comment]
@fronugget Thoughts on this?

File: paymentrails.gemspec [hunk 1]
```diff
@@ -12,5 +12,7 @@ Gem::Specification.new do |s|
   s.files = Dir.glob ["README.rdoc", "LICENSE", "lib/**/*.{rb,crt}", "spec/**/*", "*.gemspec"]
   s.required_ruby_version = '>= 2.4'
   s.add_dependency "rest-client", ">= 2.0.0"
+  s.add_development_dependency 'rake', '~> 12'
   s.add_development_dependency "rubocop", '~> 0.77'
+  s.add_development_dependency 'test-unit', '~> 3'
```
Comment (maschwenk) [inline paymentrails.gemspec:17]: Ruby has gone through the process of gem-ifying things that were formerly part of core, so that's why this is here

**Comment by fronugget** [comment]
All cool. Thanks again. 
At some point I'd like to have key & secret environment variables or as test config variables, and also should maybe think about deleting test records after the test (to avoid a huge sandbox env)

**Comment by maschwenk** [comment]
@fronugget we have the same problem developing features for ourselves, it's hard for devs to collaborate because we could be stepping on the toes of other devs doing testing

**Pull request merged**